omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyResource2_0.java

```diff
@@ -0,0 +1,246 @@
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_9.PatientResource1_9;
+
+@SubResource(parent = PatientResource1_9.class, path = "allergy", supportedClass = Allergy.class, supportedOpenmrsVersions = {
+        "2.0.*" })
+public class PatientAllergyResource2_0 extends DelegatingSubResource<Allergy, Patient, PatientResource1_9> {
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			// what should be in default representation
+			description.addProperty("display");
+			description.addProperty("uuid"); // always null, allergy does not support uuid
+			description.addProperty("allergyId");
+			description.addProperty("allergen");
+			description.addProperty("severity");
+			description.addProperty("comment");
+			description.addProperty("reactions");
+			description.addProperty("patient");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			// what should be in full representation
+			description.addProperty("display");
+			description.addProperty("uuid"); // always null, allergy does not support uuid
+			description.addProperty("allergyId");
+			description.addProperty("allergen");
+			description.addProperty("severity");
+			description.addProperty("comment");
+			description.addProperty("reactions");
+			description.addProperty("patient");
+			description.addProperty("voided");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		//description.addProperty("uuid"); // always null, allergy does not support uuid
+		description.addProperty("allergyId");
+		description.addProperty("allergen");
+		description.addProperty("severity");
+		description.addProperty("comment");
+		description.addProperty("reactions");
+        return description;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getUpdatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		return getCreatableProperties();
```

Comment:
Allergen can be changed if for example an error was made during creation, otherwise allergy has to be deleted and recreated.